src/protocols/device_path_to_text.rs

```diff
@@ -0,0 +1,30 @@
+//! Device Path to Text Protocol
+//!
+//! Convert device nodes and paths to text.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x8b843e20,
+    0x8132,
+    0x4852,
+    0x90,
+    0xcc,
+    &[0x55, 0x1a, 0x4e, 0x4a, 0x7f, 0x1c],
+);
+
+pub type DevicePathToTextNode = eficall! {fn(
+    *const crate::protocols::device_path::Protocol,
+    crate::base::Boolean,
+    crate::base::Boolean,
+) -> *const crate::base::Char16};
+
+pub type DevicePathToTextPath = eficall! {fn(
+    *const crate::protocols::device_path::Protocol,
```

Comment:
I converted this to `*mut` now, to be the same as the other protocols. We never applied `*const` annotations to function parameters so far, since they have been violated a lot in UEFI implementations, and it just makes rust miscompile when adding safe-abstractions that rely on those annotations. We always assumed `*mut` so far, and left it to safe abstractions to evaluate whether it actually requires mutable state, or whether they can fake the mut-pointer.

I am open to re-evaluating our position, but I want this as a separate discussion, not as part of this protocol addition. Note that the HII interfaces slipped through review and have some `*const` annotations. But that's mostly for historical reasons. If we want this, I think we should check whether they make sense and actually document where and when we use it.

Note that regarding UB both `*const` and `*mut` are equivalent in Rust. In fact, there have been thoughts on just providing one type of raw pointers in rust. The only real difference of both regarding FFI is their coercion from `&v` and `&mut v`. You can get `*const v` from `&v` automatically, but not from `&mut v`. In the latter case you explicitly need to as-cast from `*const` to `*mut`, which is completely legal and allowed.

Reply:
Yeah, I think a mutable pointer is better here. After some looking around, it seems the spec here means it's a pointer to a constant (which is different from a const pointer) according to the spec. I will open a separate issue where this can be discussed.